charts/ingress-nginx/templates/controller-service-internal.yaml

```
@@ -15,7 +15,7 @@ metadata:
   name: {{ include "ingress-nginx.controller.fullname" . }}-internal
   namespace: {{ include "ingress-nginx.namespace" . }}
 spec:
-  type: "{{ .Values.controller.service.type }}"
+  type: "{{ .Values.controller.service.internal.type }}"
```

Can you leave the value for `controller.service.internal.type` empty and fallback to `controller.service.type` by using the following expression:

```suggestion
  type: "{{ .Values.controller.service.internal.type | default .Values.controller.service.type }}"
```

This makes your change backwards compatible for people who changed `controller.service.type` so both their Services are using the specified type.